config/notifiers.go

```diff
@@ -188,6 +200,41 @@ func (c *SlackConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	return checkOverflow(c.XXX, "slack config")
 }
 
+// HipchatConfig configures notifications via Hipchat.
+type HipchatConfig struct {
+	NotifierConfig `yaml:",inline"`
+
+	APIURL        string `yaml:"api_url"`
+	AuthToken     Secret `yaml:"auth_token"`
+	RoomID        string `yaml:"room_id"`
+	From          string `yaml:"from"`
+	Notify        string `yaml:"notify"`
+	Message       string `yaml:"message"`
+	MessageFormat string `yaml:"message_format"`
+	Color         string `yaml:"color"`
+
+	// Catches all undefined fields and must be empty after parsing.
+	XXX map[string]interface{} `yaml:",inline"`
+}
+
+// UnmarshalYAML implements the yaml.Unmarshaler interface.
+func (c *HipchatConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	*c = DefaultHipchatConfig
+	type plain HipchatConfig
+	if err := unmarshal((*plain)(c)); err != nil {
+		return err
+	}
+	if c.RoomID == "" {
+		return fmt.Errorf("missing room id in Hipchat config")
+	}
+
+	if c.AuthToken == "" {
```

Comment:
For Slack the webhook URL already contains the authentication information and can be set in the `global` section. For PagerDuty for example that makes little sense as you have different service keys that influence the destination of notifications.

For Slack and Hipchat that is determined via channel/room though, so I imagine one would rarely have more than one token/key/etc.

Should we make this settable via the `global` section?